pgx/src/datum/date.rs

```diff
@@ -8,54 +8,79 @@ Use of this source code is governed by the MIT license that can be found in the
 */
 
 use crate::{pg_sys, FromDatum, IntoDatum};
-use std::ops::{Deref, DerefMut};
-use time::format_description::FormatItem;
+use std::ffi::CStr;
 
 #[derive(Debug)]
```

Comment:
Oh, also I think we can at least provide equality tests.
```suggestion
#[derive(Debug, PartialEq)]
```

Reply:
I went with `#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]` because they should be comparable, sortable, etc.  I'm having doubts about `Ord` though is that only or string based data?

Reply:
`Ord` is for data which exposes a **total** order. Thus there is no `Ord` implementation for `f32`, as `NAN != NAN` and `-0.0 == 0.0`, where both break the total order.

As long as `#[derive(PartialOrd)]` is correct, then implementing Ord is fine.

Reply:
I needed `Eq` to make this match work
``` 
match self {
    &Self::NEG_INFINITY | &Self::INFINITY => {
          pg_sys::EncodeSpecialDate(self.0, buf);
    ...
}
```
If `Ord` doesn't hurt i'll leave it in there.